tests/qchem/test_tapering.py

```diff
@@ -629,3 +630,209 @@ def test_taper_obs(symbols, geometry, charge):
+                @ qml.utils.sparse_hamiltonian(particle_num_tapered)
+                @ scipy.sparse.coo_matrix(excited_state_tapered).T
+            ).toarray()
+            assert np.isclose(pnum_val, pnum_val_tapered)
+
+
+@pytest.mark.parametrize(
+    ("operation", "gen_op", "message_match"),
+    [
+        (qml.U2(1, 1, 2), None, "is not implemented, please provide it with 'gen_op' args"),
+        (
+            qml.U2(1, 1, 2),
+            np.identity(16),
+            "Generator for the operation needs to a qml.Hamiltonian",
+        ),
+        (
+            qml.U2(1, 1, 2),
+            qml.Hamiltonian([1], [qml.Identity(2)]),
+            "doesn't seem to be the correct generator for the",
+        ),
+    ],
+)
+def test_inconsistent_taper_ops(operation, gen_op, message_match):
+    r"""Test that an error is raised if a set of inconsistent arguments is input"""
+
+    symbols, geometry, charge = (
+        ["He", "H"],
+        np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.4588684632]]),
+        1,
+    )
+    mol = qml.qchem.Molecule(symbols, geometry, charge)
+    hamiltonian = qml.qchem.diff_hamiltonian(mol)(geometry)
+
+    generators = qml.symmetry_generators(hamiltonian)
+    paulixops = qml.paulix_ops(generators, len(hamiltonian.wires))
+    paulix_sector = optimal_sector(hamiltonian, generators, mol.n_electrons)
+    wire_order = hamiltonian.wires
+
+    with pytest.raises(Exception, match=message_match):
+        taper_operation(operation, generators, paulixops, paulix_sector, wire_order, gen_op)
+
+
+@pytest.mark.parametrize(
+    ("operation", "gen_op"),
+    [
+        (qml.PauliX(1), qml.Hamiltonian((-np.pi / 2,), [qml.PauliX(wires=[1])])),
+        (qml.PauliZ(3), qml.Hamiltonian((-np.pi / 2,), [qml.PauliZ(wires=[3])])),
+        (
+            qml.OrbitalRotation(1, wires=[0, 1, 2, 3]),
+            qml.Hamiltonian(
+                (0.25, -0.25, 0.25, -0.25),
+                [
+                    qml.PauliX(wires=[0]) @ qml.PauliY(wires=[2]),
+                    qml.PauliY(wires=[0]) @ qml.PauliX(wires=[2]),
+                    qml.PauliX(wires=[1]) @ qml.PauliY(wires=[3]),
+                    qml.PauliY(wires=[1]) @ qml.PauliX(wires=[3]),
+                ],
+            ),
+        ),
+    ],
+)
+def test_consistent_taper_ops(operation, gen_op):
+    r"""Test that operations are tapered consistently when their generators are provided manually and when they are constructed internally"""
+
+    symbols, geometry, charge = (
+        ["He", "H"],
+        np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.4588684632]]),
+        1,
+    )
+    mol = qml.qchem.Molecule(symbols, geometry, charge)
+    hamiltonian = qml.qchem.diff_hamiltonian(mol)(geometry)
+
+    generators = qml.symmetry_generators(hamiltonian)
+    paulixops = qml.paulix_ops(generators, len(hamiltonian.wires))
+    paulix_sector = optimal_sector(hamiltonian, generators, mol.n_electrons)
+    wire_order = hamiltonian.wires
```

Comment:
Just a suggestion, feel free to ignore. We could probably also remove this and have the true results stored directly (with a comment mentioning how the tapered operation was computed).